RFS/src/main/java/org/opensearch/migrations/bulkload/transformers/Transformer.java

```diff
@@ -16,5 +18,5 @@ public interface Transformer {
     /**
      * Takes the raw JSON representing the Index Metadata of one version and returns a new, transformed copy of the JSON
      */
-    public IndexMetadata transformIndexMetadata(IndexMetadata indexData);
+    public List<IndexMetadata> transformIndexMetadata(IndexMetadata indexData);
```

Comment:
See the comment above about the composability problem.  It's hard to get one's head around this once you know that we need to chain some transforms together.

What if you keep all of the "simple transforms" that are IndexMetadata->IndexMetadata.  For when you need to return a list or take a list in, use an adapter that does the wrapping and any fanout.  It feels capricious/fragile in the code below when we're boxing within a list.